btree/btree.c

```diff
@@ -3865,17 +3875,21 @@ static int64_t btree_put_kv_tick(struct m0_sm_op *smop)
 			}
 		}
 		/** Fall through if path_check is successful. */
+	case P_DECIDE_NXT: {
+		if (bop->bo_opc == M0_BO_PUT) {
+			if (oi->i_key_found)
+				return P_ACT;
+			else
+				return  P_MAKESPACE;
+		} else {
+			M0_ASSERT(bop->bo_opc == M0_BO_UPDATE);
+			return P_ACT;
+		}
```

Comment:
Remove these lines and move the code at lines 3917 - 3929 here with appropriate changes.

Reply:
done.